File: python/56.py


def digital_sum(n):
    return sum(map(int, list(str(n))))

if __name__ == "__main__":
    a, b = 100, 100
    maximum = 0
    for i in range(1, a):
        for j in range(1, b):
            M = digital_sum(i ** j)
            if M > maximum:
                maximum = M

    print(maximum)

File: cpp/56.cpp
#include <iostream>
#include <string>
#include "bigint.cpp"

using namespace std;

int digital_sum(string s)
{
    int sum_{};
    for (char c : s)
        sum_ += c - '0';
    return sum_;
}

int main()
{
    int maximum {}, M {};

    for (int i = 1; i < 100; i++)
    {
        for (int j = 1; j < 100; j++)
        {
            Dodecahedron::Bigint a = i;
            M = digital_sum(to_string(a.pow(j)));
            if (M > maximum)
                maximum = M;
        }
    }

    cout << maximum << endl;

    return 0;
}